marklogic-data-hub/src/main/java/com/marklogic/hub/impl/DataHubImpl.java

```diff
@@ -376,7 +376,8 @@ public void clearUserModules() {
             );
 
             String query =
-                "cts:uris((),(),cts:not-query(cts:collection-query('hub-core-module')))[\n" +
+                "cts:uris((),(),cts:not-query(cts:collection-query(('hub-core-module','http://marklogic.com/entity-services/mapping'" +
```

Comment:
This doesn't really seem safe... I think if a user runs "mlClearModulesDatabase", the user would expect the xml/xslt files generated for their own custom functions to be deleted as well. 

This seems to be addressing a timing issue in QS, right? Where between the time it takes for QS to save the mapping and then reinstall modules, the user may choose to run their mapping? 

In which case - isn't their always a chance that the user's custom mapping functions are deleted and not yet reloaded by the time the user chooses to run the mapping?

I'll play around with this locally. My hunch is that in the QS context, we don't need to clear the modules database in the first place. Clearing modules makes sense when you e.g. start your day and pull from git and you want to ensure any deletions in git are made in your modules database as well. I don't think it's necessary to do while in QS though.